persistence/src/main/scala/hmda/persistence/institutions/InstitutionPersistence.scala

```diff
@@ -27,8 +27,13 @@ object InstitutionPersistence {
         case InstitutionCreated(i) =>
           InstitutionsState(institutions + i)
         case InstitutionModified(i) =>
-          val x = institutions.find(x => x.id == i.id).getOrElse(Institution())
-          InstitutionsState((institutions - x) + i)
+          if (institutions.exists(x => x.id == i.id)) {
+            val x = institutions.find(x => x.id == i.id).get
+            InstitutionsState((institutions - x) + i)
+          } else {
+            InstitutionsState(institutions + i)
```

Comment:
I suspect this line is untested. (Check codecov results when available.)

I'm also not too happy about iterating over the list twice and using `get`.  Are you aiming to improve readability?  If so, I suggest:
```
          val others: Set[Institution] = institutions.filterNot(_.id == i.id)
          InstitutionsState(others + i)
```
We may eventually want to use something that supports fast+easy lookup by id, too.

No strong preference about the `x => x.id` vs `_.id` here, by the way.